check_diff/src/lib.rs

```diff
@@ -270,3 +347,36 @@ pub fn compile_rustfmt(
         feature_runner,
     });
 }
+
+fn search_for_rs_files(repo: &Path) -> impl Iterator<Item = PathBuf> {
+    return WalkDir::new(repo)
+        .into_iter()
+        .filter_map(|e| e.ok())
+        .filter(|entry| {
+            let path = entry.path();
+            return path.is_file() && path.extension().map_or(false, |ext| ext == "rs");
+        })
+        .map(|entry| entry.into_path());
+}
```

Comment:
Really happy to see that this has been abstracted out. Is there a way to combine these chained calls to `filter_map`, `filter`, and `map` into a single `filter_map` call?